### PR TITLE
Clean up base Command and CommandResponse APIs

### PR DESCRIPTION
**Pull request by labkey-tchad**
#### Rationale
Started some code cleanup against a feature branch. Re-targeting to develop.

#### Related Pull Requests
* #38

#### Changes
* Remove `Command.copy` and unnecessary copy constructors
* Make Response constructors more consistent
* Make `ResponseObject` properties immutable
* Remove long unused `org.labkey.remoteapi.security.ACL`
* Chip away at missing JavaDoc

### CODE REVIEW

File: labkey-client-api/CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # The LabKey Remote API Library for Java - Change Log
 
-## version 4.0.0
+## version 4.0.0-jsonUpgrade-SNAPSHOT
```
Comment (labkey-adam) [inline labkey-client-api/CHANGELOG.md]: Is this intentional?

File: labkey-client-api/build.gradle [hunk 2]
```diff
@@ -48,7 +49,7 @@ repositories {
 
 group "org.labkey.api"
 
-version "4.0.0-SNAPSHOT"
+version "4.0.0-jsonUpgrade-SNAPSHOT"
```
Comment (labkey-adam) [inline labkey-client-api/build.gradle]: Is this intentional?

**Review by labkey-adam — CHANGES_REQUESTED** [review]
Please summarize these changes in `CHANGELOG.md`

Note that this still builds with 100+ JavaDoc warnings, but perhaps that's for another day...

File: labkey-client-api/CHANGELOG.md (hunk 1 above)
Comment (labkey-tchad) [reply labkey-client-api/CHANGELOG.md]: Sorry, meant to make this a draft. Waiting on results from TeamCity. I'll re-request review once TeamCity has taken its pass.

File: labkey-client-api/CHANGELOG.md [hunk 3]
```diff
@@ -2,11 +2,15 @@
 
 ## version 4.0.0
 *Released*: TBD
-* Replace the internal JSON library with [JSON-java](https://github.com/stleary/JSON-java). The previous library,
+* Migrade to a new JSON library: [JSON-java](https://github.com/stleary/JSON-java). The previous library,
```
Comment (labkey-adam) [inline labkey-client-api/CHANGELOG.md]:
```suggestion
* Migrate to a new JSON library: [JSON-java](https://github.com/stleary/JSON-java). The previous library,
```

File: labkey-client-api/CHANGELOG.md [hunk 4]
```diff
@@ -2,11 +2,15 @@
 
 ## version 4.0.0
 *Released*: TBD
```
Comment (labkey-adam) [inline labkey-client-api/CHANGELOG.md]:
```suggestion
*Released*: 26 October 2022
```

File: labkey-client-api/build.gradle [hunk 5]
```diff
@@ -48,7 +48,7 @@ repositories {
 
 group "org.labkey.api"
 
-version "4.0.0-SNAPSHOT"
+version "4.0.0-codeCleanup-SNAPSHOT"
```
Comment (labkey-adam) [inline labkey-client-api/build.gradle]:
```suggestion
version "4.1.0-SNAPSHOT"
```

**Review by labkey-adam — CHANGES_REQUESTED** [review]
Couple tweaks requested so we're prepped for publishing today and next round of development

**Review by labkey-adam — APPROVED** [review]

**Pull request merged**